### PR TITLE
Disappearing Back Button Fix

### PR DESCRIPTION
**Pull request by mattbarker016**

### CODE REVIEW

**Comment by mattbarker016** [comment]
Also added way to remove buses no longer being live tracked. Some codebase quality improvements, etc.

File: TCAT/Views/BusLocationView.swift [hunk 1]
```diff
@@ -98,16 +96,18 @@ class BusLocationView: UIView {
             }
         }
         
+        let newDegrees = Double(degrees) - self.currentBearing
```
Comment (myo3) [inline TCAT/Views/BusLocationView.swift]: this is kind of nit-picky but do you need the `self.`?

**Comment by myo3** [comment]
looks good to me!

File: TCAT/Views/BusLocationView.swift (hunk 1 above)
Comment (mattbarker016) [reply TCAT/Views/BusLocationView.swift]: I look putting a little bit of my`self` into my code

Comment (mattbarker016) [reply TCAT/Views/BusLocationView.swift]: It was a leftover from being in the `UIView.animate` closure

Comment (myo3) [reply TCAT/Views/BusLocationView.swift]: lol okay then yeah `self` is necessary

Comment (myo3) [reply TCAT/Views/BusLocationView.swift]: edit: misread your comment but cool 👍

**Pull request merged**